simple-factsheet.Rmd

```diff
@@ -23,91 +28,401 @@ library(tidyverse)
 library(knitr)
 library(kableExtra)
 library(urbnthemes)
+library(here)
+library(scales)
+library(readxl)
+library(stringr)
+library(ggrepel)
 
 set_urbn_defaults(style = "print")
 ```
 
+```{r plots, include=FALSE, results="hide"}
+filepath <- here("/data/factsheet_who are-support_data_states_201718_varnames.xlsx")
+
+topercent <- function(x) {
+  if(abs(x) > 0.01) {
+    round((x * 100),0)
+  } else{
+    round((x * 100),1)
+  }
+} 
+
+state_text <- ifelse(params$state == "District of Columbia",
+                     paste("the",params$state,sep=" "),
+                     params$state)
+
+states_subset <- read_excel(filepath) %>%
+  filter(StateName == params$state) %>%
+  rename_at(vars(contains("1718")),~sub("_1718", "", .)) %>%
+  
+  # prepare overall cOI population number - rounded thousand or numeric + "million" text
+  mutate(text_1_popcount = case_when(total_popcount_coi < 1000000 ~ formatC(total_popcount_coi,
+                                                                            format ="f", 
+                                                                            digits = 0,
+                                                                            big.mark =","),
+                                     TRUE ~ paste(round((total_popcount_coi/1000000),1), "million"))) %>%
+  
+  # convert decimals to be communicated as percentages - no decimals except under 1% 
+  # Except for those that will only be used in data vis 
+  mutate_at(vars(starts_with("pct"),
+                 starts_with("share")), topercent) %>%
+  
+  # prepare strings that'll change
+  mutate(text_2_citizens = case_when(share_uscitizen_coi >= 85 ~ "Almost all",
+                                     share_uscitizen_coi < 85 & share_uscitizen_coi >= 50 ~ "Most",
+                                     share_uscitizen_coi < 50 ~ "Many",
+                                     TRUE ~ NA_character_),
```

Comment:
Don't add `TRUE` to `case_when()` to add `NA`. The `NA` will implicitly be added to all observations that don't meet the conditions.